app/lib/submission_builder/document.rb

```diff
@@ -99,6 +99,11 @@ def add_non_zero_value(xml, elem_name, line)
       end
     end
 
+    def add_element_if_present(xml, tag, line_id)
+      value = calculated_fields.fetch(line_id)
+      xml.send(tag, value) if value.present?
+    end
```

Comment:
if `value` is not present here, it won't `send` to the xml which means that `TaxDue` node will _not_ exist? (because it hasn't been "added"? Or does that node already exist but the value is not being filled? If latter, i'd rename the method to something like `fill_element_if_present`

Reply:
i see this was a method that already existed -- i'd still love to know if I understand this correctly though!

Reply:
@arinchoi03 Yes to the first. If the value doesn't exist (is nil), then it won't create the element and add it to the XML